src/base/utils/net.cpp

```diff
@@ -89,5 +90,25 @@ namespace Utils
         {
             return subnet.first.toString() + '/' + QString::number(subnet.second);
         }
+
+        boost::optional<QList<QSslCertificate>> tryLoadSSLCertificate(const QByteArray &data)
```

Comment:
I would omit wrapping it in `boost::optional`. I don't like this approach. I would either trow an error or just return empty list.

Reply:
>I don't like this approach.

Why? Compared with your suggestions the only overhead of `optional` is 1 byte bool and the code calling this function is easier to read.

>I would either trow an error

This is a bit overkill, we are calling this function directly not through a chain of function calls, i.e. we can/should check its error status directly.

>just return empty list.

Then the client code (such as in optionsdialog.cpp) will be forced to know about the type `<T>` of `optional<T>` which is unnecessary for whom only wish to know success/failure of the function call.

Reply:
The traditional way would have a signature like `bool loadSSLCertificate(const QByteArray &data, QList<QSslCertificate> *outputList)`, without going into the details you can already see how messy its internal is going to be compared to the current code (and also the code using it).

Reply:
>Why? Compared with your suggestions the only overhead of optional is 1 byte bool  is 1 byte bool and the code calling this function is easier to read.

"Class template optional is a wrapper for representing 'optional' (or 'nullable') objects who may not (yet) contain a valid value."
Both result types already have this feature (can represent its "nullability" or "emptyness") so wrapping it in `boost::optional` is the real overkill.

>This is a bit overkill, we are calling this function directly not through a chain of function calls, i.e. we can/should check its error status directly.

I don't know how directly calling a function conflicts with being able to throw an exception...
In addition, a particular use case should not affect either the implementation of the function or its interface. Interface/implementation should reflect only its meaning/job.

>Then the client code (such as in optionsdialog.cpp) will be forced to know about the type <T> of optional<T> which is unnecessary for whom only wish to know success/failure of the function call.

This is not a valid reason to use such a workaround.

In fact, all I offered was to get rid of the double parsing. I didn't think it would drive your mind into such a jungle.
The solution is simpler, it will provide readable code. What we need? To load the SSL key:
```c++
QSslKey loadSSLKey(const QByteArray &data)
{
    // try different formats
    QSslKey key {data, QSsl::Rsa};
    if (key.isNull())
        return QSslKey(data, QSsl::Ec);
}
```
You can use it where you need its result (e.g. in Server code).
If you need only check its result without propagating type info to the calling code you just need a simple helper:
```c++
bool isValidSSLKey(const QByteArray &data)
{
    return !loadSSLKey(data).isNull();
}
```

Reply:
>Both result types already have this feature (can represent its "nullability" or "emptyness") so wrapping it in boost::optional is the real overkill.

This is a valid point, I agree with it more or less.

>If you need only check its result without propagating type info to the calling code you just need a simple helper

Yes, however the way I see it, `optional` is just a convenient interface that we get to write less code, with only a low price to pay.

I'm not going to insist anymore so I'll add the helper function.
PR updated.